sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.VisitStmt.cs

```diff
@@ -391,9 +391,10 @@ private void VisitCXXUuidofExpr(CXXUuidofExpr cxxUuidofExpr)
 
         private void VisitDeclRefExpr(DeclRefExpr declRefExpr)
         {
-            if ((declRefExpr.Decl is EnumConstantDecl enumConstantDecl) && (declRefExpr.DeclContext != enumConstantDecl.DeclContext) && (enumConstantDecl.DeclContext is NamedDecl namedDecl))
+            if ((declRefExpr.Decl is EnumConstantDecl { DeclContext: NamedDecl namedDecl } enumConstantDecl) &&
```

Comment:
nit: I really don't think this pattern is more readable than the `&&` expression.
It's likely a point in time thing due to patterns being less widespread, but at least right now, this is much harder for me to parse and understand what is being checked.